vmm/src/lib.rs

```diff
@@ -1342,18 +1337,39 @@ impl Vmm {
 
         self.init_guest_memory()?;
 
-        self.setup_interrupt_controller()?;
+        let vcpus;
+
+        #[cfg(target_arch = "x86_64")]
+        {
+            self.setup_interrupt_controller()?;
+
+            self.attach_virtio_devices()?;
+            self.attach_legacy_devices()?;
+
+            let entry_addr = self.load_kernel()?;
+
+            self.configure_system()?;
 
-        self.attach_virtio_devices()?;
-        self.attach_legacy_devices()?;
+            self.register_events()?;
 
-        let entry_addr = self.load_kernel()?;
+            vcpus = self.create_vcpus(entry_addr, request_ts)?;
+        }
 
-        self.configure_system()?;
+        #[cfg(target_arch = "aarch64")]
+        {
+            let entry_addr = self.load_kernel()?;
 
-        self.register_events()?;
+            vcpus = self.create_vcpus(entry_addr, request_ts)?;
 
-        let vcpus = self.create_vcpus(entry_addr, request_ts)?;
+            self.setup_interrupt_controller()?;
+
+            self.attach_virtio_devices()?;
+            self.attach_legacy_devices()?;
+
+            self.register_events()?;
```

Comment:
Why are register_events and configure_system swapped?

Reply:
No reason and I also just realized that there is no reason in having them inside the aarch64 cfg. I removed them from there. Take a look.